### PR TITLE
CATROID-1121 Refactor MainMenuActivity and related files to Kotlin

### PR DESCRIPTION
**Pull request by khushal1707**
TIcket: https://jira.catrob.at/browse/CATROID-1121
- Refactored MainMenuActivity, TermsOfUseDialogFragment, AboutDialogFragment to kotlin
- Applied View Binding to above mentioned file

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s gitflow workflow
- [x] Verify that your changes do not have any conflicts with the base branch
- [x] After the PR, verify that all CI checks have passed
- [ ] Post a message in the *catroid-stage* or *catroid-ide* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

File: catroid/src/main/java/org/catrobat/catroid/ui/MainMenuActivity.kt [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+        val termsOfUseUrlStringText = getString(R.string.main_menu_terms_of_use)
+        val termsOfUseUrl = getString(
+            R.string.terms_of_use_link_template,
+            Constants.CATROBAT_TERMS_OF_USE_URL +
+                Constants.CATROBAT_TERMS_OF_USE_TOKEN_FLAVOR_URL + BuildConfig.FLAVOR +
+                Constants.CATROBAT_TERMS_OF_USE_TOKEN_VERSION_URL + BuildConfig.VERSION_CODE,
+            termsOfUseUrlStringText
+        )
+
+        termsOfUseUrlTextView.text = Html.fromHtml(termsOfUseUrl)
```
Comment (ndrnour) [inline catroid/src/main/java/org/catrobat/catroid/ui/MainMenuActivity.kt:118]:
```suggestion
        termsOfUseUrlTextView.text = if (Build.VERSION.SDK_INT >= 24) {
            Html.fromHtml(termsOfUseUrl, HtmlCompat.FROM_HTML_MODE_LEGACY)
        } else {
            Html.fromHtml(termsOfUseUrl)
        }
```

File: catroid/src/main/java/org/catrobat/catroid/ui/MainMenuActivity.kt [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+
+        AlertDialog.Builder(this)
+            .setNegativeButton(R.string.decline) { dialog: DialogInterface?, which: Int -> handleDeclinedPrivacyPolicyButton() }
+            .setPositiveButton(R.string.accept) { dialog: DialogInterface?, which: Int -> handleAgreedToPrivacyPolicyButton() }
```
Comment (ndrnour) [inline catroid/src/main/java/org/catrobat/catroid/ui/MainMenuActivity.kt:122]:
```suggestion
.setNegativeButton(R.string.decline) { _, _ -> handleDeclinedPrivacyPolicyButton() }
.setPositiveButton(R.string.accept) { _, _ -> handleAgreedToPrivacyPolicyButton() }
```

File: catroid/src/main/java/org/catrobat/catroid/ui/MainMenuActivity.kt [hunk 3]
```diff
@@ -0,0 +1,356 @@
...
+            .setCancelable(false)
+            .setOnKeyListener { dialog: DialogInterface?, keyCode: Int, event: KeyEvent? ->
```
Comment (ndrnour) [inline catroid/src/main/java/org/catrobat/catroid/ui/MainMenuActivity.kt:124]:
```suggestion
            .setOnKeyListener { _, keyCode: Int, _ ->
```

File: catroid/src/main/java/org/catrobat/catroid/ui/MainMenuActivity.kt [hunk 4]
```diff
@@ -0,0 +1,356 @@
...
+
+        val linkString = getString(
+            R.string.about_link_template,
+            Constants.BASE_APP_URL_HTTPS,
+            getString(R.string.share_website_text)
+        )
+
+        val linkTextView = declinedTermsOfUseViewBinding.shareWebsiteView
+        linkTextView.movementMethod = LinkMovementMethod.getInstance()
+        linkTextView.text = Html.fromHtml(linkString)
```
Comment (ndrnour) [inline catroid/src/main/java/org/catrobat/catroid/ui/MainMenuActivity.kt:161]:
```suggestion
        linkTextView.text = if (Build.VERSION.SDK_INT >= 24) {
            Html.fromHtml(linkString, HtmlCompat.FROM_HTML_MODE_LEGACY)
        } else {
            Html.fromHtml(linkString)
        }
```

File: catroid/src/main/java/org/catrobat/catroid/ui/MainMenuActivity.kt [hunk 5]
```diff
@@ -0,0 +1,356 @@
...
+
+        AlertDialog.Builder(this)
+            .setView(dialogView)
+            .setPositiveButton(R.string.ok) { dialog: DialogInterface?, which: Int -> showTermsOfUseDialog() }
```
Comment (ndrnour) [inline catroid/src/main/java/org/catrobat/catroid/ui/MainMenuActivity.kt:165]:
```suggestion
            .setPositiveButton(R.string.ok) { _, _ -> showTermsOfUseDialog() }
```

File: catroid/src/main/java/org/catrobat/catroid/ui/MainMenuActivity.kt [hunk 6]
```diff
@@ -0,0 +1,356 @@
...
+            .setCancelable(false)
+            .setOnKeyListener { dialog: DialogInterface, keyCode: Int, event: KeyEvent? ->
```
Comment (ndrnour) [inline catroid/src/main/java/org/catrobat/catroid/ui/MainMenuActivity.kt:167]:
```suggestion
           .setOnKeyListener { dialog: DialogInterface, keyCode: Int, _ ->
```

File: catroid/src/main/java/org/catrobat/catroid/ui/MainMenuActivity.kt [hunk 7]
```diff
@@ -0,0 +1,356 @@
...
+            webIntent.putExtra(WebViewActivity.INTENT_PARAMETER_URL, shareUri.toString())
+            startActivity(webIntent)
+        }
+    }
+
+    private fun setShowProgressBar(show: Boolean) {
+        val progressBarBinding = ProgressBarBinding.inflate(layoutInflater)
+        progressBarBinding.root.visibility = if (show) View.VISIBLE else View.GONE
+        mainMenuBinding.fragmentContainer.visibility =
+            if (show) View.GONE else View.VISIBLE
```
Comment (ndrnour) [inline catroid/src/main/java/org/catrobat/catroid/ui/MainMenuActivity.kt:224]:
```suggestion
       progressBarBinding.root.setVisibleOrGone(show)
       mainMenuBinding.fragmentContainer.setVisibleOrGone(!show)
```

File: catroid/src/main/java/org/catrobat/catroid/ui/MainMenuActivity.kt [hunk 8]
```diff
@@ -0,0 +1,356 @@
...
+
+    override fun onCreateOptionsMenu(menu: Menu): Boolean {
+        menuInflater.inflate(R.menu.menu_main_menu, menu)
+        val scratchConverter = getString(R.string.main_menu_scratch_converter)
+        val scratchConverterBeta = SpannableString(
+            scratchConverter + " " + getString(R.string.beta)
+        )
+
+        scratchConverterBeta.setSpan(
+            ForegroundColorSpan(resources.getColor(R.color.beta_label_color)),
```
Comment (ndrnour) [inline catroid/src/main/java/org/catrobat/catroid/ui/MainMenuActivity.kt:244]:
```suggestion
            ForegroundColorSpan(resources.getColor(R.color.beta_label_color, theme)),
```

File: catroid/src/main/java/org/catrobat/catroid/ui/dialogs/TermsOfUseDialogFragment.kt [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+
+        termsOfUseDialogBuilder.setPositiveButton(R.string.ok) { dialog, id -> dialog.dismiss() }
+
+        termsOfUseTextView.setText(R.string.dialog_terms_of_use_info)
+
+        val termsOfUseUrl = getString(
+            R.string.terms_of_use_link_template, Constants.CATROBAT_TERMS_OF_USE_URL,
+            termsOfUseUrlStringText
+        )
+        termsOfUseUrlTextView.text = Html.fromHtml(termsOfUseUrl)
```
Comment (ndrnour) [inline catroid/src/main/java/org/catrobat/catroid/ui/dialogs/TermsOfUseDialogFragment.kt:62]:
```suggestion
        termsOfUseUrlTextView.text = if (Build.VERSION.SDK_INT >= 24) {
            Html.fromHtml(termsOfUseUrl, HtmlCompat.FROM_HTML_MODE_LEGACY)
        } else {
            Html.fromHtml(termsOfUseUrl)
        }
```

File: catroid/src/main/java/org/catrobat/catroid/ui/dialogs/TermsOfUseDialogFragment.kt [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+        termsOfUseUrlTextView.movementMethod = LinkMovementMethod.getInstance()
+
+        val termsOfUseUrlStringText = getString(R.string.dialog_terms_of_use_link_text)
+        val termsOfUseDialogBuilder = AlertDialog.Builder(
+            requireActivity()
+        )
+            .setView(binding.root)
+            .setTitle(R.string.dialog_terms_of_use_title)
+
+        termsOfUseDialogBuilder.setPositiveButton(R.string.ok) { dialog, id -> dialog.dismiss() }
```
Comment (ndrnour) [inline catroid/src/main/java/org/catrobat/catroid/ui/dialogs/TermsOfUseDialogFragment.kt:54]:
```suggestion
        termsOfUseDialogBuilder.setPositiveButton(R.string.ok) { dialog, _ -> dialog.dismiss() }
```

File: catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/dialog/AboutDialogFragment.kt [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+        )
+        developerUrlView.text = Html.fromHtml(developerUrl)
+
+        val aboutCatrobatView = binding.dialogAboutTextCatrobatUrl
+        aboutCatrobatView.movementMethod = LinkMovementMethod.getInstance()
+        val aboutCatrobatUrl = getString(
+            R.string.about_link_template, Constants.CATROBAT_ABOUT_URL,
+            getString(R.string.dialog_about_catrobat_link_text)
+        )
+        aboutCatrobatView.text = Html.fromHtml(aboutCatrobatUrl)
```
Comment (ndrnour) [inline catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/dialog/AboutDialogFragment.kt:59]: same

File: catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/dialog/AboutDialogFragment.kt [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+    override fun onCreateDialog(bundle: Bundle?): Dialog {
+        _binding = DialogAboutBinding.inflate(LayoutInflater.from(activity))
+
+        val developerUrlView = binding.dialogAboutTextViewUrl
+        developerUrlView.movementMethod = LinkMovementMethod.getInstance()
+        val developerUrl = getString(
+            R.string.about_link_template, Constants.ABOUT_POCKETCODE_LICENSE_URL,
+            getString(R.string.dialog_about_license_link_text)
+        )
+        developerUrlView.text = Html.fromHtml(developerUrl)
```
Comment (ndrnour) [inline catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/dialog/AboutDialogFragment.kt:51]: same

**Review by ndrnour — CHANGES_REQUESTED** [review]
please consider the suggested changes in your Code.

**Comment by ndrnour** [comment]
also please fix this [APK FAILED](https://jenkins.catrob.at/job/Catroid/job/PR-4209/16/execution/node/39/log/)
`21:16:04  > Task :catroid:compileStandaloneDebugKotlin
21:16:04  e: /home/catroid/jenkins_slave/workspace/Catroid_PR-4209/catroid/src/main/java/org/catrobat/catroid/ui/BaseActivity.kt: (40, 49): Unresolved reference: surveyCampaign
21:16:04  e: /home/catroid/jenkins_slave/workspace/Catroid_PR-4209/catroid/src/main/java/org/catrobat/catroid/ui/BaseActivity.kt: (97, 9): Unresolved reference: surveyCampaign
21:16:04  e: /home/catroid/jenkins_slave/workspace/Catroid_PR-4209/catroid/src/main/java/org/catrobat/catroid/ui/BaseActivity.kt: (150, 13): Unresolved reference: surveyCampaign
21:16:04  e: /home/catroid/jenkins_slave/workspace/Catroid_PR-4209/catroid/src/main/java/org/catrobat/catroid/ui/BaseActivity.kt: (151, 13): Unresolved reference: surveyCampaign
21:16:04  
21:16:04  > Task :catroid:compileStandaloneDebugKotlin FAILED`

**Comment by khushal1707** [comment]
Done!

**Comment by ndrnour** [comment]
please keep an eye on #4286, once it gets merged, rebase and then I'll review it

**Review by bengaal — APPROVED** [review]
Looks good to me!

**Review by DinosaurierRex — APPROVED** [review]
PO review LGTM 👍 

@Walcho1125 or @juliajulie95 Please merge! Thnks

**Pull request merged**